cpp/src/io/json/json_tree.cu

```diff
@@ -61,7 +61,10 @@ struct token_to_node {
       case token_t::StructBegin: return NC_STRUCT;
       case token_t::ListBegin: return NC_LIST;
       case token_t::StringBegin: return NC_STR;
-      case token_t::ValueBegin: return NC_VAL;
+      case token_t::ValueBegin:
+        return NC_STR;  // NC_VAL;
+      // NV_VAL is removed because type inference and
+      // reduce_to_column_tree category collapsing takes care of this.
```

Comment:
This feels like a comment for the reviewer, and not future readers of the code. Perhaps:
```suggestion
      case token_t::ValueBegin:
        return NC_STR;  // type inference and reduce_to_column_tree category collapsing will later convert this to a value
```

WDYT? (I don't know the details of the type inference, so my proposed comment might be wrong)